### DISCUSSION TITLE
2021-09-05 (automatic release)

### DISCUSSION THREAD

**Discussion by pipcet** [pipcet/pearl#18] [Announcements]
https://github.com/pipcet/pearl/releases/tag/2021-09-05

This release boots to the pearl screen, then, after a minute, boots a Debian system running off a ramfs. Note that as it is not backed by persistent storage, any changes you make and data you create won't persist across reboots.

Instructions to follow in a second post (which hopefully won't be eaten by github as the first draft was).

**Comment by pipcet** [comment]
Step-by-step instructions. Please let me know if there are any questions, problems, comments, etc!

1. Download https://github.com/pipcet/pearl/releases/download/2021-09-05/pearl-debian.macho
2. Install `pearl-debian.macho` as custom kernel
3. connect a USB keyboard to one of the Type-C ports. The Type-A ports might or might not work.
4. reboot
5. if all goes well, you'll see the pearl boot screen with a menu in the upper left quadrant. Ignore it and wait for a minute.
6. There should be some output as Debian is configured and runs. Wait for a shell prompt.
7. Log in as `root`, password `x`.
8. if desired. switch to 4k mode by running
```
cd /usr
tar xvf /linux.modules
depmod -a
modprobe apple
```
9. Configure your network interface using, for example, `nmtui`.
10. Send a success report!

**Reply by Johnnie390** [reply]
How to login as root? Password=x ? Sorry, I have lost you.

**Reply by Johnnie390** [reply]
linux.modules is now in /persist and /boot.
Ran this - 
cd /usr
tar xvf /linux.modules
depmod -a
modprobe apple
Screenshot follows.
![Modprobe apple](https://user-images.githubusercontent.com/22815859/132124215-2b9fa6e1-dcbd-4ff5-a0d0-cac6e6ecf426.jpeg)

nmtui not available at this stage as I have no desktop.

Ry

**Reply by Johnnie390** [reply]
What now? I am missing my desktop....

Ry

**Reply by Johnnie390** [reply]
Spoke too soon / was impatient.

Now logged in as root (Password = x). Lsmod report apple module loaded.

**Reply by Johnnie390** [reply]
Desktop present (LXDE). Everything looks very small (4K) from the CLI. Success. Well done!!

Ry

**Reply by cristim** [reply]
Awesome! Would this work on the M1 MBP or only on the mini for now?

**Reply by Johnnie390** [reply]
No idea.. ask Pip.

Good luck

PS This "experiment" is non-persistent over reboots - for the moment.

**Reply by pipcet** [reply]
@cristim wrote:
> Awesome! Would this work on the M1 MBP or only on the mini for now?

My main machine is an M1 MBP so it should work on those. The problem is wifi, which requires firmware which is currently available only by copying it from macOS. USB dongles should work (connect them before you boot, though), too.

I'm currently looking at automation for the copy-wifi-firmware step, which will hopefully also entail writing instructions for how to install pearl.

If you're willing to test, that would be great! What kind of hardware do you have?

And, yes, @Johnnie390 is right, you need to explicitly `debootstrap` to a fixed disk to have it work there, for the moment. It's doable, but a bit difficult.

**Reply by cristim** [reply]
I  have the M1 MBP, that's why I asked about it 😁

How about other drivers? Keyboard, touchbar, display over thunderbolt?

**Reply by pipcet** [reply]
> I have the M1 MBP, that's why I asked about it grin

Ah, okay. I'm about to post new instructions for the release that is being built right now, and it does work the way it should on my MBP.

> How about other drivers? Keyboard, touchbar, display over thunderbolt?

Keyboard works. The touchbar seems difficult to get to work and we're still in the initial stages of understanding how it does work. Display over thunderbolt is something I've been meaning to look at but the cable/adapter I got doesn't appear to be working even with macOS.

**Reply by cristim** [reply]
Thanks, I'll try it out over the next few days and report back any issues.

**Reply by pipcet** [reply]
> Thanks, I'll try it out over the next few days and report back any issues.

Thanks, that would be awesome! Do let me know if there's anything I can help with.
